### PR TITLE
Support different encodings in Image.toByteData()

### PR DESCRIPTION
**Pull request by tvolkert**
Fixes https://github.com/flutter/flutter/issues/16635

### CODE REVIEW

**Comment by Hixie** [comment]
What is a use case motivating the unmodified feature?

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -1217,20 +1249,20 @@ class Image extends NativeFieldWrapperClass2 {
 
   /// Converts the [Image] object into a byte array.
   ///
-  /// The image bytes will be RGBA form, 8 bits per channel, row-primary.
+  /// [format] specifies the format in which the bytes will be returned.
```
Comment (Hixie) [inline lib/ui/painting.dart]: avoid starting with a lowercase letter by saying something like "The [...] argument ..."

**Comment by tvolkert** [comment]
> What is a use case motivating the unmodified feature?

Performance when you're dealing with non-RGBA source images.  It stemmed from a discussion with @chinmaygarde

**Comment by tvolkert** [comment]
flutter/flutter#16859

**Comment by lukehutch** [comment]
Hi, I'm commenting here since https://github.com/flutter/flutter/issues/16635 is closed.

All of the target platforms have native codecs for JPEG and other image and video formats. So why not expose at least some minimum subset of these?

JPEG compression at the very least absolutely needs to be supported in Flutter. The 3rd party JPEG libraries in pub.dev are either very slow, or very hacky, and/or only support encoding and decoding from PNG (which requires an additional compression step), and/or only support JPEG-to-JPEG resizing (or have other restrictions).

I would love to see Flutter having a built-in API for some very basic image processing operations: compress/decompress, resize, rotate, and crop. I'm a little shocked that other than for `ResizeImage`, these basic operations are not supported without some fairly involved code (drawing onto a `Canvas` and using `Picture` and `Recorder` to create an image).

For video, in addition to these basic image processing operations, there also needs to be a built-in API to trim the video.

**Pull request merged**